### PR TITLE
CDAP-15818: Run-transform plugin validation.

### PR DESCRIPTION
**Pull request by AntonGensitskiy**
JIRA: https://issues.cask.co/browse/CDAP-15818
WIKI: https://wiki.cask.co/display/CE/Guidelines+on+Plugin+Validation

In scope of this PR:

- integrated with new validation API

### CODE REVIEW

File: src/main/java/io/cdap/plugin/Run.java [hunk 1]
```diff
@@ -59,24 +55,20 @@ public Run(RunConfig config) {
...
     Schema inputSchema = stageConfigurer.getInputSchema();
-    Preconditions.checkArgument(inputSchema != null, "Input Schema must be a known constant.");
-    config.validateBinaryExecutableType();
-    config.validateInputFields(inputSchema);
-    config.validateOutputFieldType();
-    if (!Strings.isNullOrEmpty(config.schema)) {
-      config.verifyOutputFieldTypeInSchema();
-    }
+    FailureCollector failureCollector = stageConfigurer.getFailureCollector();
+    config.validate(failureCollector, inputSchema);
```
Comment (mlozbin-cybervisiontech) [inline src/main/java/io/cdap/plugin/Run.java:59]: you forgot to check failure collector

File: src/test/java/io/cdap/plugin/RunConfigTest.java [hunk 2]
```diff
@@ -17,73 +17,104 @@
...
+
     try {
-      new Run(config).configurePipeline(configurer);
-      Assert.fail();
-    } catch (IllegalArgumentException e) {
-      Assert.assertEquals("Error while accessing the binary. Please make sure that the 'Command to Execute' is in " +
-                            "the expected format. 'java -jar'", e.getMessage());
+      config.validate(failureCollector, inputSchema);
+    } catch (ValidationException e) {
+      assertValidationFailed(failureCollector, paramName);
```
Comment (mlozbin-cybervisiontech) [inline src/test/java/io/cdap/plugin/RunConfigTest.java]: this test will not fail if validation passes but it should

File: src/test/java/io/cdap/plugin/RunConfigTest.java [hunk 3]
```diff
@@ -17,73 +17,104 @@
...
+
     try {
-      new Run(config).configurePipeline(configurer);
-      Assert.fail();
-    } catch (IllegalArgumentException e) {
-      Assert.assertEquals("Binary type 'dll' is not supported. Supported executable types are: 'exe, sh, bat and jar'" +
-                            ".", e.getMessage());
+      config.validate(failureCollector, inputSchema);
+    } catch (ValidationException e) {
+      assertValidationFailed(failureCollector, paramName);
```
Comment (mlozbin-cybervisiontech) [inline src/test/java/io/cdap/plugin/RunConfigTest.java]: this test will not fail if validation passes but it should

File: src/test/java/io/cdap/plugin/RunConfigTest.java [hunk 4]
```diff
@@ -17,73 +17,104 @@
...
+    List<List<String>> paramName = Collections.singletonList(
+      Collections.singletonList(RunConfig.COMMAND_TO_EXECUTE));
+
     try {
-      new Run(config).configurePipeline(configurer);
-      Assert.fail();
-    } catch (IllegalArgumentException e) {
-      Assert.assertEquals("Binary type 'dll' is not supported. Supported executable types are: 'exe, sh, bat and jar'" +
-                            ".", e.getMessage());
+      config.validate(failureCollector, inputSchema);
```
Comment (mlozbin-cybervisiontech) [inline src/test/java/io/cdap/plugin/RunConfigTest.java]: validate method should not throw validation exception

**Review by mlozbin-cybervisiontech — CHANGES_REQUESTED** [review]

File: src/main/java/io/cdap/plugin/Run.java (hunk 1 above)
Comment (AntonGensitskiy) [reply src/main/java/io/cdap/plugin/Run.java:59]: I not added failure collector check  based on the discussion with Vinisha
https://github.com/data-integrations/topn/pull/12#discussion_r323842824

File: src/test/java/io/cdap/plugin/RunConfigTest.java (hunk 2 above)
Comment (AntonGensitskiy) [reply src/test/java/io/cdap/plugin/RunConfigTest.java]: FIxed

File: src/test/java/io/cdap/plugin/RunConfigTest.java (hunk 3 above)
Comment (AntonGensitskiy) [reply src/test/java/io/cdap/plugin/RunConfigTest.java]: Fixed

File: src/test/java/io/cdap/plugin/RunConfigTest.java (hunk 4 above)
Comment (AntonGensitskiy) [reply src/test/java/io/cdap/plugin/RunConfigTest.java]: Done

**Pull request closed without merging**